tests/unittests/BackendTest.cpp

```diff
@@ -146,6 +146,32 @@ TEST_P(BackendTest, debugPrint) {
   function->execute();
 }
 
+/// This test checks that we can compile a function without depending on the IR.
+/// We compile some function and then delete the function. Later we execute the
+/// code and check that things work.
+TEST_P(BackendTest, decoupleCodegenFromIR) {
+  Module mod;
+  Function *F = mod.createFunction("main");
+  auto *X = mod.createVariable(ElemKind::FloatTy, {3}, "X");
+  X->getPayload().getHandle() = {1., 2., 3.};
+  auto *pow = F->createPow("Pow1", X, 2.0);
+  auto *save = F->createSave("save", pow);
+  Variable *res = save->getVariable();
+  EE_.compile(CompilationMode::Infer, F);
+
+  // Erase all of the functions to ensure that the compiled code does not
+  // depend on the IR.
```

Comment:
Do you mean "graph representation" when you say "IR" here? Our interpreter or OpenCL cannot run without the low-level IR. But they can run without the graph.